iroha_actor/src/broker.rs

```diff
@@ -234,155 +159,11 @@ pub trait BrokerMessage: Message<Result = ()> + Clone + 'static + Send {}
 impl<M: Message<Result = ()> + Clone + 'static + Send> BrokerMessage for M {}
 
 #[cfg(test)]
+#[allow(clippy::unwrap_used)]
 mod tests {
-    use async_trait::async_trait;
-    use tokio::sync::RwLock;
-
     use super::*;
 
-    pub struct Actor1(Broker, Arc<RwLock<u32>>);
-
-    #[async_trait]
-    impl Actor for Actor1 {
-        async fn on_start(&mut self, ctx: &mut Context<Self>) {
-            self.0.subscribe::<StopMessage, _>(ctx);
-            self.0.subscribe::<Message1, _>(ctx);
-        }
-
-        fn broker(&self) -> Option<&broker::Broker> {
-            Some(&self.0)
-        }
-    }
-
-    #[derive(Clone, Copy)]
-    pub struct StopMessage;
-
-    impl Message for StopMessage {
-        type Result = ();
-    }
-
-    #[async_trait::async_trait]
-    impl ContextHandler<StopMessage> for Actor1 {
-        type Result = ();
-
-        async fn handle(
-            &mut self,
-            ctx: &mut Context<Self>,
-            StopMessage: StopMessage,
-        ) -> Self::Result {
-            ctx.stop_now()
-        }
-    }
-
-    #[derive(Clone, Copy)]
-    pub struct Message1;
-
-    impl Message for Message1 {
-        type Result = ();
-    }
-
-    #[async_trait::async_trait]
-    impl Handler<Message1> for Actor1 {
-        type Result = ();
-
-        async fn handle(&mut self, Message1: Message1) -> Self::Result {
-            *self.1.write().await += 1;
-        }
-    }
-
-    #[tokio::test]
-    #[allow(clippy::unwrap_used)]
-    async fn actor_unsubscribes_on_stop() {
```

Comment:
Why have you removed this test and this feature?

Reply:
Made another test which incorporates both of those

Reply:
Can you show me where?

Reply:
Usually it is better not to rework the tests, but to keep them the same so that we know that new PR doesn't break anything

Reply:
https://github.com/hyperledger/iroha/pull/1333/files#diff-939764a3f6ff31d1cc59be8347d73178bb2139a59c122f7d84a4313be6314452R182